archive_envoy_release.sh

```
@@ -76,7 +76,7 @@ not_yet_version() {
   echo >&2 "version ${version} not yet supported on ${os}-${arch}" && exit 1
 }
 
-curl="curl -sSL"
+curl="curl -fsSL"
```

lack of -f meant we didn't notice this was silently passing on fail also